src/txmempool.cpp

```diff
@@ -1150,3 +1276,112 @@ size_t CTxMemPool::DynamicMemoryUsage() const {
           cachedInnerUsage);
 }
 
+std::pair<uint256, CAmount> CTxMemPool::FindConflictingCert(const uint256& scId, int64_t certQuality)
+{
+    LOCK(cs);
+    std::pair<uint256, CAmount> res = std::make_pair(uint256(),CAmount(-1));
+
+    if (mapSidechains.count(scId) == 0)
+        return res;
+
+    for(const auto& mempoolCertEntry : mapSidechains.at(scId).mBackwardCertificates)
+    {
+        const CScCertificate& mempoolCert = mapCertificate.at(mempoolCertEntry.second).GetCertificate();
+        if (mempoolCert.quality == certQuality) {
+            res.first  = mempoolCert.GetHash();
+            res.second = mapCertificate.at(mempoolCertEntry.second).GetFee();
+            break;
+        }
+    }
+
+    return res;
+}
+
+bool CTxMemPool::RemoveConflictingQualityCert(const uint256& certToRmHash)
+{
+    LOCK(cs);
+
+    if(mapCertificate.count(certToRmHash) == 0)
+        return true; //nothing to remove
+
+    CScCertificate certToRm = mapCertificate.at(certToRmHash).GetCertificate();
+    std::list<CTransaction> conflictingTxs;
+    std::list<CScCertificate> conflictingCerts;
+    remove(certToRm, conflictingTxs, conflictingCerts, true);
+
+    // the conflicting cert itself must be the only one in this list
+    assert(conflictingCerts.size() == 1); //this assert should prolly be done in mempool::check
+
+    // Tell wallet about transactions and certificates that went from mempool to conflicted:
+    for(const auto &t: conflictingTxs) {
+        LogPrint("mempool", "%s():%d - syncing tx %s\n", __func__, __LINE__, t.GetHash().ToString());
+        SyncWithWallets(t, nullptr);
+    }
+    for(const auto &c: conflictingCerts) {
+        LogPrint("mempool", "%s():%d - syncing cert %s\n", __func__, __LINE__, c.GetHash().ToString());
+        SyncWithWallets(c, nullptr);
+    }
+
+    return true;
+}
+
+bool CTxMemPool::IsTopQualityCertInMempool(const CScCertificate& cert)
```

Comment:
It is actually the top quality check for the certificates for given `cert.scId`. Let's think about more accurate name.

Reply:
Actually not called anymore. Dropped